### PR TITLE
ok this is a huge diff

### PR DESCRIPTION
**Pull request by benharri**
i was bored at lunch today and it was staring at me from my editor.

* code formatted with [black](/ambv/black)
* connect, join, and other common stuff is now in util.py
* it's probably broken

### CODE REVIEW

**Comment by RussellChamp** [comment]
:eyes:

**Comment by benharri** [comment]
from what i can tell, the bots connect and run fine (tested locally with an oragono ircd). there might be a few things left over though with the str.format() changes i did.

File: Code/irc/tildebot.py [hunk 1]
```diff
@@ -220,21 +209,17 @@ def get_prize(name, isHuman, bonus=0):
...
+                    ),
                 ]
             else:  # hit jackpot!
                 jackpotfile.write(str(JACKPOT_MIN))
                 return [
                     jackpot,
-                    name
-                    + " hit the jackpot and won **"
-                    + p.number_to_words(jackpot)
-                    + " tildes!**",
```
Comment (RussellChamp) [inline Code/irc/tildebot.py]: forgot "tildes!"

File: Code/irc/tildebot.py [hunk 2]
```diff
@@ -344,7 +324,8 @@ def rollcall(channel):
 
 
 def connect(server, channel, botnick):
```
Comment (RussellChamp) [inline Code/irc/tildebot.py]: connect method should be removed if the util.connect method is being used

File: Code/irc/wangbot.py [hunk 3]
```diff
@@ -75,64 +72,41 @@ def resetGlobals():
...
-def joinchan(chan):
-    ircsock.send("JOIN " + chan + "\n")
-
-
-def hello():
-    ircsock.send("PRIVMSG " + channel + " :Hello!\n")
-
-
 def start_numberwang(channel, user):
     if channel != "#bots":
```
Comment (RussellChamp) [inline Code/irc/wangbot.py:76]: i should probably change this to be the passed in channel instead of hard coded

File: Code/irc/dict_puzzle.py [hunk 4]
```diff
@@ -34,14 +32,12 @@ def get_puzzle():
...
     dict_words = [
-        word for word in get_wordlist() if len(word) > 2 and len(word) <= maxlen + 1
+        word for word in gen_wordlist() if len(word) > 2 and len(word) < maxlen + 1
     ]
     word = random.choice(dict_words)
-    while True:
-        anagram = "".join(random.sample(word, len(word)))
-        if anagram != word:
-            break
-    puzzle = "Unscramble the following word: '" + anagram + "'"
```
Comment (RussellChamp) [inline Code/irc/dict_puzzle.py:34]: what happened to "puzzle = ..." ?

File: Code/irc/tildebot.py [hunk 5]
```diff
@@ -48,7 +48,7 @@
 SCORE_FILE = "tildescores.txt"
 JACKPOT_FILE = "tildejackpot.txt"
 JACKPOT_MIN = 3
-DEBUG = False
+DEBUG = True
```
Comment (RussellChamp) [inline Code/irc/tildebot.py]: no

File: Code/irc/tildescores.txt [hunk 6]
```diff
@@ -122,14 +122,4 @@ x4464&^%1&^%1532028546
 pounce&^%19&^%1532133325
 von_&^%5&^%1532502104
 livix&^%7&^%1533603142
-ben&^%3&^%1533767627
```
Comment (RussellChamp) [inline Code/irc/tildescores.txt:125]: please undo this

File: Code/irc/topicscores.txt [hunk 7]
```diff
@@ -1 +1,2 @@
 vilmibm&^%0&^%2
```
Comment (RussellChamp) [inline Code/irc/topicscores.txt]: this one, too

File: Code/irc/util.py [hunk 8]
```diff
@@ -4,30 +4,44 @@
...
 
 def connect(ircsock, options):
-    server, channel, botnick = options
-    server, port = server.split(":")
-    ircsock.connect((server, port))
-    ircsock.send("USER {0} {0} {0} :krowbar".format(botnick))
-    ircsock.send("NICK {}\r\n".format(botnick))
-    ircsock.send("MODE +B {}\r\n".format(botnick))
-    joinchan(channel)
+    print(options)
```
Comment (RussellChamp) [inline Code/irc/util.py:31]: were these intentionally left?

Comment (benharri) [reply Code/irc/util.py:31]: options isn't a tuple

File: Code/irc/dict_puzzle.py (hunk 4 above)
Comment (benharri) [reply Code/irc/dict_puzzle.py:34]: looks like it got renamed to `challenge_text`

**Comment by RussellChamp** [comment]
work code is compiling... time to review ~town PRs

**Pull request merged**